src/gui/options_banlist.cpp

```diff
@@ -0,0 +1,100 @@
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+#include "options_banlist.h"
+#include "ui_options_banlist.h"
+
+#include <QMessageBox>
+#include <libtorrent/address.hpp>
+
+#include "base/bittorrent/session.h"
+
+options_banlist::options_banlist(QWidget *parent)
+    : QDialog(parent)
+    , m_ui(new Ui::BanListOptions)
+    , m_modified(false)
+{
+    m_ui->setupUi(this);
+    m_ui->bannedIPList->addItems(BitTorrent::Session::instance()->getBannedIPList());
+    m_ui->pb_BanIP->setEnabled(false);
+}
+
+options_banlist::~options_banlist()
+{
+    delete m_ui;
+}
+
+void options_banlist::on_buttonBox_accepted()
+{
+    if (m_modified) {
+        // save to session
+        QStringList IPList;
+        for (int i = 0; i < m_ui->bannedIPList->count(); ++i)
+            IPList << m_ui->bannedIPList->item(i)->text();
+        BitTorrent::Session::instance()->setBannedIPList(IPList);
+    }
+    accept();
+}
+
+bool options_banlist::checkInputIP(const QString& ip)
+{
+    if (ip.isEmpty()) return false;
+    boost::system::error_code ec;
+    libtorrent::address::from_string(ip.toLatin1().constData(), ec); // only error code needed
+    if (ec) return false;
+    return true;
+}
+
+void options_banlist::on_pb_BanIP_clicked()
+{
+    QString s = m_ui->ed_IP->text();
+    if (s.isEmpty()) {
+        QMessageBox::warning(this, tr("Warning"), tr("Entered IP address is invalid."));
+        return;
+    }
+    m_ui->bannedIPList->addItem(s);
+    m_ui->ed_IP->clear();
+    m_modified = true;
+}
+
+void options_banlist::on_pb_DeleteIP_clicked()
+{
+    QList<QListWidgetItem *> selection = m_ui->bannedIPList->selectedItems();
+    for (auto &i : selection) {
+        m_ui->bannedIPList->removeItemWidget(i);
+        delete i;
+    }
+    m_modified = true;
+}
+
+void options_banlist::on_ed_IP_textChanged(const QString &arg1)
```

Comment:
1. renamed arg1 to ip. 
2. QRegExpValidator does not fits good here. Custom validator will be just more complicated wrapping around `libtorrent::address::from_string` than my code.
Also i dislike UI designs when input field suddenty drop my input when it does not pass intermediate validator check. It is enough that action button is disabled while there is wrong value in input, but edit is not blocked. 

Use case: I copy IP that I want to ban from somewhere and also copied some letters to the left of IP. (May be I was in a hurry, or there is selection problems.) I want to paste it into qbittorent, cut off wrong letters and ban it, but ta-da - validator does not allow me to paste because is will be wrong value, and it force me to use intermediate text editor paste-strip-copy steps. 

Also don't forget about maintaining complicated custom validator code, and such principles like KISS and YAGNI